src/secret_sharing/decomposed.rs

```diff
@@ -39,6 +39,24 @@ impl<S> BitDecomposed<S> {
     pub fn map<F: Fn(S) -> T, T>(self, f: F) -> BitDecomposed<T> {
         BitDecomposed::new(self.bits.into_iter().map(f))
     }
+
+    pub fn len(&self) -> usize {
+        return self.bits.len();
+    }
+
+    /// Convert this into a base-10 field value.
+    /// The inner vector of this type is a list any field type (e.g. Z2, Zp) and
+    /// each element is (should be) a share of 1 or 0. This function iterates
+    /// over the shares of bits and computes Σ(2^i * b_i).
+    pub fn as_decimal<F>(&self) -> S
```

Comment:
I'm going to quibble with you on the name of this one. Decimal implies base ten. We aren't using base ten. Can we call it something like: to_additive_sharing_in_large_field().